[Bug]: All `npx storybook@latest ...` commands fail with `Error: Cannot find module './util'`

### Describe the bug

After getting this error when trying the `npx storybook@latest init` command on my NextJS project, I cleared NPM/NPX caches, and also tried on a brand new blank project, and it failed with the same message each time.

```
[me@myhost:~/tmp]$ npx storybook@latest init
Need to install the following packages:
  storybook@7.0.23
Ok to proceed? (y) y
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module './util'
Require stack:
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/common/index.js
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/FileSystemCache.js
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/index.js
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/core-common/dist/index.js
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/telemetry/dist/index.js
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/cli/dist/generate.js
- /home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/cli/bin/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/common/index.js:30:27)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/common/index.js',
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/FileSystemCache.js',
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/file-system-cache/lib/index.js',
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/core-common/dist/index.js',
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/telemetry/dist/index.js',
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/cli/dist/generate.js',
    '/home/me/.npm/_npx/bc7e1e37fcb46ffc/node_modules/@storybook/cli/bin/index.js'
  ]
}

Node.js v18.16.1
```

### To Reproduce

`npx storybook@latest init`

### System

```shell
> Please paste the results of npx storybook@latest info here.

I get the same error when running that.

- Linux (Ubuntu 22.04.2)
- Node v18.16.1
- NPM v9.5.1
```


### Additional context

_No response_

--- a/code/lib/core-common/package.json
+++ b/code/lib/core-common/package.json
@@ -52,7 +52,7 @@
     "chalk": "^4.1.0",
     "esbuild": "^0.18.0",
     "esbuild-register": "^3.4.0",
-    "file-system-cache": "^2.0.0",
+    "file-system-cache": "2.3.0",
     "find-cache-dir": "^3.0.0",
     "find-up": "^5.0.0",
     "fs-extra": "^11.1.0",

--- a/code/lib/types/package.json
+++ b/code/lib/types/package.json
@@ -46,7 +46,7 @@
     "@storybook/channels": "7.1.0-alpha.39",
     "@types/babel__core": "^7.0.0",
     "@types/express": "^4.7.0",
-    "file-system-cache": "^2.0.0"
+    "file-system-cache": "2.3.0"
   },
   "devDependencies": {
     "@storybook/csf": "^0.1.0",

